### DISCUSSION TITLE
How to use WineHQ releases?

### DISCUSSION THREAD

**Discussion by asmaloney** [The-Wineskin-Project/WineskinServer#432] [Q&A] [answered]
Is there a way to try WineHQ releases instead of the built-in CrossOver ones?

It looks like there's [a bug fix in 7.8](https://bugs.winehq.org/show_bug.cgi?id=52845) that is required for software which uses CEF (Chromium Embedded Framework). I think the CrossOver releases are based on an earlier version of Wine - 7.1 or something?

I'm currently unable to run Guild Wars 2 because it now requires CEF and I'm hoping that might be the fix.

Thanks!

**Edit:** Note that the answer solves the problem I was facing, not the question I asked 😄

**Comment by Gcenx** [comment]
You could just replace the wrapper “wine” directory contents with one of my Winehq builds.

Though upstream doesn’t have all the nice hacks that CX does so something won’t work.

**Reply by Gcenx** [reply]
I’m assuming your trying to run the official client launcher instead of from Steam?

I could look into integrating this patch but CX23 beta is currently happening (beta2) so we’ll get sources shortly.

**Reply by asmaloney** [reply]
> I’m assuming your trying to run the official client launcher instead of from Steam?

Yes. Unfortunately the standalone client and the Steam one are not interchangeable, so accounts created with one won't work on the other.

> I could look into integrating this patch but CX23 beta is currently happening (beta2) so we’ll get sources shortly.

Cool. I'm not 100% sure it's the only thing to fix the issue.

As of the GW2 patch on Tuesday (18th), it is unplayable on macOS. There are (at least) two things that are intertwined:

- it launches _without DXVK_, however it's unplayable at 9-10 FPS (before the patch, running with DXVK and working around CEF, it ran at 45-50 FPS)
- launching _with DXVK_ does not work because CEF is now required (no workaround) and CEF doesn't work with DXVK

Had a hard time finding logs to provide further info, so I went looking for CEF/DXVK issues and found the one above...

**Reply by Gcenx** [reply]
So your saying everything works as long as your not using DXVK the game just runs awfully?

**Reply by asmaloney** [reply]
Well sort-of, yeah. It's slow and doesn't look as good. I even turned all the settings to the lowest and it is still unplayable. Can't get more than around 10-12 FPS.

What would it be using for DX11 if DXVK isn't installed?

**Reply by Gcenx** [reply]
When DXVK isn’t installed its using WineD3D Vulkan.

Though since you didn’t mention launcher issues when not using DXVK there might be a solution, I’ll need to download the standalone launcher first to verify.

**Reply by asmaloney** [reply]
Here are the instructions I wrote up (and have been updating) for other people to use: https://asmaloney.com/2022/07/howto/running-guild-wars-2-with-dx11-on-macos-using-wine/

With DXVK installed it now gets stuck "Initializing" when you try to launch the game.

**Reply by Gcenx** [reply, marked as answer]
You could try the following

- Open winecfg
- Libraries tab
- dxgi (set to builtin)

Let the wine process exit then try to run the game again, that might get you up and running again.

**Reply by Gcenx** [reply]
There will already be a dxgi override set too (native,builtin) just change it to (builtin)

**Reply by asmaloney** [reply]
That... looks like it works! This is amazing. (I'll be able to test more thoroughly later, but it launches and I seem to be getting reasonable FPS with some hitching.)

Thank you so much Dean - GW2 players can live another day!

But... why does it work?

(Moved because I threaded this comment incorrectly.)

**Reply by Gcenx** [reply]
I’ve seen this before with other things but from what I understand is the application queries dxgi for the levels and decides to use d3d9, the problem is DXVK dxgi only works with its own d3d9 dll (Vulkan based) not wines dll that is currently only usable via OpenGL.

As the base game is free I’ll take a look at including a builtin fix.

**Reply by asmaloney** [reply]
> As the base game is free I’ll take a look at including a builtin fix.

That would be amazing - thanks!

In the meantime I'll update my instructions.

**Comment by weijiangan** [comment]
@Gcenx do your CX engines have any modifications on top of CX? I recently tried running wine-staging directly on my machine and it seems to run my DX9 game much slower than in Wineskin

**Reply by Gcenx** [reply]
> do your CX engines have any modifications on top of CX?

Many changes.

> I recently tried running wine-staging directly on my machine and it seems to run my DX9 game much slower than in Wineskin

There’s still some CX hacks that haven’t been upstreamed, I’ve recently started go apply a small number of them to make upstream at least functional under Rosetta2

**Reply by weijiangan** [reply]
hey thanks for the prompt response. I tried looking for the source of the engines but I'm having some troubles finding them, are they open source? Otherwise would love to know on a high level what kind of changes have you made on top of CX. Don't get me wrong I used Wineskin for ages but recently I just felt like I want to know what goes into actually making Wine run well. 

I'm trying to run an old DX9 game that didn't run well at all under vanilla Wine and even it ran well in Wineskin there are some bugs and I'm trying to dive into it and learning how to debug it

**Reply by Gcenx** [reply]
> I tried looking for the source of the engines but I'm having some troubles finding them, are they open source?

The sources aren’t public due to where some of these patchsets originate.

> Otherwise would love to know on a high level what kind of changes have you made on top of CX.

It’s a mix of things:
- reverting select CXhacks
- cherry picked upstream commits
- custom hacks

WineskinCX23.6.0 is 72 commits on-top, there could have been more with more free time to allocate to this.

> Don't get me wrong I used Wineskin for ages but recently I just felt like I want to know what goes into actually making Wine run well.

You’d want to join the Winehq mailing lists and also check winehq gitlab instance.

> I'm trying to run an old DX9 game that didn't run well at all under vanilla Wine and even it ran well in Wineskin there are some bugs and I'm trying to dive into it and learning how to debug it

If something doesn’t run properly with wine-devel-8.21 I’m sure Zeb and the other WineD3D maintainers would love a Winehq bug.
